django/db/backends/oracle/creation.py

```diff
@@ -216,18 +223,14 @@ def _create_test_user(self, cursor, parameters, verbosity):
                      CREATE TRIGGER
                TO %(user)s""",
         ]
-        self._execute_statements(cursor, statements, parameters, verbosity)
+        acceptable_ora_err = None
+        if keepdb:
+            # Avoid user already exists error when keepdb is on
+            acceptable_ora_err = 'ORA-01920'
+        self._execute_allow_fail_statements(cursor, statements, parameters, verbosity, acceptable_ora_err)
         # Most test-suites can be run without the create-view privilege. But some need it.
         extra = "GRANT CREATE VIEW TO %(user)s"
-        try:
-            self._execute_statements(cursor, [extra], parameters, verbosity, allow_quiet_fail=True)
-        except DatabaseError as err:
-            description = str(err)
-            if 'ORA-01031' in description:
-                if verbosity >= 2:
-                    print("Failed to grant CREATE VIEW permission to test user. This may be ok.")
```

Comment:
I can assign return value to ``_execute_allow_fail_statements`` (``false`` - if fail, ``true`` if properly), then in line ``233``:  
```python
result = self._execute_allow_fail_statements(cursor, [extra], parameters, verbosity, 'ORA-01031')
if not result and verbosity >= 2:
    print("Failed to grant CREATE VIEW permission to test user. This may be ok.")
```